### PR TITLE
remove legacy installation of Roxygen2 5.0 and add R-specific clean target (#12993)

### PR DESCRIPTION
**Pull request by shadogray**
## Description ##
Remove the forceful installation of legacy Roxygen2 version.
Add an R-specific clean target (rclean).

## Checklist ##
see Makefile, R-package/R/zzz.R

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage due to install test during build and rpkttest target
- [x] To the my best knowledge, examples are not affected by this change

### Changes ###
- [x] rpkg uses less legacy and proprietary mechanisms - still not aligned with standard R
- [x] rclean target extracts R-specific tasks from common clean target

## Comments ##
- the packaging is still _very_ awkward and should be revised

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -584,14 +584,15 @@ rpkg:
 	cp -rf 3rdparty/tvm/nnvm/include/* R-package/inst/include
 	Rscript -e "if(!require(devtools)){install.packages('devtools', repo = 'https://cloud.r-project.org/')}"
 	Rscript -e "library(devtools); library(methods); options(repos=c(CRAN='https://cloud.r-project.org/')); install_deps(pkg='R-package', dependencies = TRUE)"
-	echo "import(Rcpp)" > R-package/NAMESPACE
-	echo "import(methods)" >> R-package/NAMESPACE
+	cp R-package/dummy.NAMESPACE R-package/NAMESPACE
```
Comment (hetong007) [inline Makefile]: Can you point me to some source on how `R CMD INSTALL` deal with `dummy.NAMESPACE`? Seems it is automatically ignored.

**Comment by ankkhedia** [comment]
@shadogray Thanks for your contribution!
Could you please take a look into review comments 

@jeremiedb @anirudhacharya Could you please help review the PR

@mxnet-label-bot [pr-awaiting-response, R ]

File: Makefile (hunk 1 above)
Comment (shadogray) [reply Makefile]: the NAMESPACE defines the exported symbols of the package. It is read on loading of package by R runtime.
To clarify the move to "dummy.NAMESPACE": the original code generated a very simple NAMESPACE (as you see in the removed lines) by pasting those 2 lines, which my have sufficed with legacy versions.
Modern build tools also check the complete package+install cycle and then detects missing symbols, so the intermediate build fails, this build is necessary to setup the infrastructure to generate the final NAMESPACE file.
This is the reason, why the CMD INSTALL is run twice, so my understanding.
Only the NAMESPACE of the second CMD INSTALL is used by final package.

**Comment by shadogray** [comment]
I tried hard to find any connex between my changes and the Python script failing, can't see any.
The above failing script relates to Anaconda/python: "c:\Anaconda3\envs\py3\Scripts\pip install", cannot see _any_ relationship with R runtime.

**Comment by shadogray** [comment]
Sorry for not having  checked the installation of missing Roxygen2 package, build on GPU and CPU have passed generation of package now.

**Comment by shadogray** [comment]
Added dummy.NAMESPACE to .Rbuildignore as requested, thanks for the comment

**Comment by ankkhedia** [comment]
@hetong007 Could you please merge this PR if it looks good?

**Pull request merged**